### ISSUE TITLE
FR: allow disabling of webcam (i.e. voice-only mode)

### ISSUE THREAD

**Issue by aspiers** [jangouts/jangouts#80]
Currently the only way to disable the webcam in Chrome is to refuse Jangouts permission to access it, but this also refuses permission to the microphone, which is obviously useless.  I would like to be able to use Jangouts in voice-only mode, even if others are using their webcams.  This is supported by Google Hangouts.

**Comment by ancorgs** [comment]
You have a button to disable your video if you mouse over your thumbnail. @aspiers is that not enough?

**Comment by aspiers** [comment]
@ancorgs Thanks for the quick reply.  No not quite enough - it needs to be possible to disable before entering the session.  I'll give a extreme example just to demonstrate the point (which BTW is a silly example and of course purely hypothetical!!!): if I joined jangouts whilst wearing no clothes, it would be too late to click the thumbnail disable video button after joining ;-)

**Comment by jreidinger** [comment]
in FF I can choose source of audio and video, so if I select video source no video, it works nice for me.

**Comment by ancorgs** [comment]
As, @jreidinger said, Firefox allows you to specify different permissions for camera and microphone. when connecting.

Chromium shows a dialog asking "allow camera and microphone?", so it merges both permissions in just one question, even if it actually recognizes both permissions as different things (if you click on the camera icon during your session or you go to "manage media settings" you'll see separate permissions lists for camera and microphone). From my point of view, that oversimplification in the interaction with the user is a Chromium issue.

There are still use cases for the "start with video disabled" checkbox, but I would consider this a low priority feature.

**Comment by aspiers** [comment]
Even if Chromium doesn't handle it well, IMHO it should still be the app's responsibility to respect the user's privacy, rather than assuming it is always OK to use the webcam.  But yes, this is not top priority.

**Comment by jreidinger** [comment]
Actually app for this case is browser, it should ensure your privacy is protected. BTW for your case does it work, to not give permission and later allow only sound?

**Comment by aspiers** [comment]
> Actually app for this case is browser, it should ensure your privacy is protected.

IMHO the browser and Jangouts are both apps, and both should ensure protection.  Relying on a single level of protection is not reliable.  The browser should protect with regard to real security threats from attackers (eavesdropping, stolen identity etc.), and Jangouts should protect in order to ensure the user's privacy is respected even when there are no attackers.

> BTW for your case does it work, to not give permission and later allow only sound?

If I don't give permission, will Jangouts still work?  Actually I just tested, and right now my camera doesn't work at all even if I want it to ...

**Comment by jreidinger** [comment]
> The browser should protect with regard to real security threats from attackers (eavesdropping, stolen  identity etc.), and Jangouts should protect in order to ensure the user's privacy is respected even when there are no attackers.

actually browser should also protect user privacy, consider that you want speak only service and browser ask you to give both permissions, so malicious code on that page can see you, what you are doing, it is not so nice.

**Comment by aspiers** [comment]
I agree. Maybe a clearer way of starting my point is that 2 lines of defence are safer than one :-)

The other issue is that it will probably take a long time to fix Chromium, so we need a short term workaround.

**Comment by jreidinger** [comment]
Workaround like not being naked when speaking with collegues? :smile:
